pyathena/util/transform.py

```diff
@@ -21,9 +21,9 @@ def euler_rotation(vec, angles):
     angles = np.array(angles)
     r = Rotation.from_euler('zyx', -angles, degrees=False)
     rotmat = r.as_matrix()
-    vxp = rotmat[0, 0]*vec[0] + rotmat[0, 1]*vec[1] + rotmat[0, 2]*vec[2]
-    vyp = rotmat[1, 0]*vec[0] + rotmat[1, 1]*vec[1] + rotmat[1, 2]*vec[2]
-    vzp = rotmat[2, 0]*vec[0] + rotmat[2, 1]*vec[1] + rotmat[2, 2]*vec[2]
+    vxp = rotmat[0, 2]*vec[2] + rotmat[0, 1]*vec[1] + rotmat[0, 0]*vec[0]
+    vyp = rotmat[1, 2]*vec[2] + rotmat[1, 1]*vec[1] + rotmat[1, 0]*vec[0]
+    vzp = rotmat[2, 2]*vec[2] + rotmat[2, 1]*vec[1] + rotmat[2, 0]*vec[0]
```

Comment:
@sanghyukmoon To me this doesn't change anything other than the order of sums. Curious how this actually helps improving performance.

Reply:
Sorry I didn't explain enough. This improves performance in a specific occasion:
`x, y, z = euler_rotation((x, y, z), [alpha, beta, 0])`
where `x`, `y`, `z` are 1-dimensional `xarray.DataArray`s along each dimension.

The resulting `x`, `y`, `z` on the left hand side denotes the Cartesian coordinates components with respect to the rotated axes (due to `euler_rotation`), which are *3-dimensional* DataArrays in terms of the original x, y, z coordinates (i.e., x' = x'(x, y, z).) In this case, the memory layout of the constructed x' is either x'(z, y, x) or x'(x, y, z) depending on the order of the operation, where the former is preferred assuming the other arrays are z, y, x order (default in Athena(++))

Reply:
The wall time spent on the multiplication of two 512^3 arrays differs by a factor of five in the un-aligned and aligned case.

Reply:
That's very interesting. Thanks very much for the clarification.